miden-tx/src/host/tx_progress.rs

```diff
@@ -0,0 +1,120 @@
+pub use alloc::vec::Vec;
+
+use miden_objects::notes::NoteId;
+
+// TRANSACTION PROGRESS
+// ================================================================================================
+
+/// Contains the information about the number of cycles for each of the transaction execution
+/// stages.
+#[derive(Default)]
+pub struct TransactionProgress {
+    prologue: CycleInterval,
+    notes_processing: CycleInterval,
+    note_execution: Vec<(Option<NoteId>, CycleInterval)>,
```

Comment:
Why is `NoteId` in `Option` here? If we try to record note execution time and there is no note ID, we should probably panic (i.e., it means there is a bug somewhere in the kernel).